critiquebrainz/db/users.py

```diff
@@ -0,0 +1,52 @@
+from hashlib import md5
+import sqlalchemy
+from critiquebrainz import db
+
+
+def avatar(user):
+    """Args:
+        user: user object from db (dict) or username string
+    Returns:
+        URL to gravatar image
+    """
+    url = "https://gravatar.com/avatar/{0}{1}"
+    if type(user) is dict:
+        if user['email'] and user['show_gravatar']:
+            return url.format(md5(user['email'].encode('utf-8')).hexdigest(),
+                              "?d=identicon&r=pg")
+        else:
+            return url.format(md5(str(user['id']).encode('utf-8')).hexdigest(),
+                              "?d=identicon")
+    return url.format(md5(user.encode('utf-8')).hexdigest(), "?d=identicon")
```

Comment:
Notice how `url.format(md5(<...>.encode('utf-8')).hexdigest(), "?d=identicon&r=pg")` is defined three times and pretty much identical. What if you just select a string to be hashed and then build a URL at the end?